okpt/io/config/validator.py

```diff
@@ -0,0 +1,109 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+"""Provides validation for various config files.
+
+Functions:
+    validate(): Validate the tool configuration.
+
+Exceptions:
+    ConfigurationError: An error in the tool configuration.
+
+"""
+
+import os
+from io import TextIOWrapper
+from typing import Any, Dict
+
+import cerberus
+
+from okpt.io import reader
+from okpt.io.config import parser
+
+_TOOL, _OPENSEARCH = ['tool', 'opensearch']
+
+
+class ConfigurationError(Exception):
+    """Exception raised for errors in the tool configuration.
+
+    Attributes:
+        message -- explanation of the error
+    """
+    def __init__(self, message: str):
+        self.message = message
+
+
+class _Validator():
```

Comment:
Why it is not public?

Reply:
Since the class isn't externally visible, I've followed the guidelines listed [here](https://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations).